modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/DLCExpiryCheckEnabledDeletionTask.java

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+
+package org.wso2.andes.kernel;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.andes.tools.utils.MessageTracer;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * DLCExpiryCheckEnabledDeletionTask responsible for deleting the expired messages from unassigned slots,
+ * and dead letter channel
+ */
+public class DLCExpiryCheckEnabledDeletionTask extends PeriodicExpiryMessageDeletionTask {
+
+    private static Log log = LogFactory.getLog(DLCExpiryCheckEnabledDeletionTask.class);
+
+    /**
+     * Get the messages form the DLC which are expired and delete those from DB
+     */
+    public void deleteExpiredMessagesFromDLC() {
+
+        try {
+            //This logic belongs to an MB run in stand alone mode / the coordinator node run in cluster mode
+            if (!isClusteringEnabled
+                    || (isClusteringEnabled && AndesContext.getInstance().getClusterAgent().isCoordinator())) {
+
+                //Checks for the message store availability if its not available
+                //Deletion task needs to await until message store becomes available
+                if (null != messageStoresUnavailable) {
+                    log.info("Message store has become unavailable therefore expiry message deletion task waiting until"
+                            + " store becomes available");
+                    //act as a barrier
+                    messageStoresUnavailable.get();
+                    log.info("Message store became available. Resuming expiry message deletion task");
+                    messageStoresUnavailable = null; // we are passing the blockade
+                    // (therefore clear the it).
+                }
+
+                //get the expired messages for that queue in the range of message ID starting form the lower bound ID
+                List<Long> expiredMessages = MessagingEngine.getInstance().getExpiredMessagesFromDLC();
+
+                if ((null != expiredMessages) && (!expiredMessages.isEmpty())) {
+
+                    //Tracing message activity
+                    if (MessageTracer.isEnabled()) {
+                        for (Long messageId : expiredMessages) {
+                            MessageTracer.trace(messageId, "", MessageTracer.EXPIRED_MESSAGE_DETECTED_FROM_DLC);
+                        }
+                    }
+
+                    //delete message metadata, content from the meta data table, content table and expiry table
+                    MessagingEngine.getInstance().deleteMessagesById(expiredMessages);
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Expired message count in DLC is :" + expiredMessages.size());
+                    }
+                }
+            }
+        } catch (AndesException e) {
+            log.error("Error running Message Expiration Checker " + e.getMessage(), e);
```

Comment:
Since this is a scheduled periodic task, we caught all the exceptions and  logged it rather than throwing out.